superset/databases/utils.py

```diff
@@ -44,7 +44,7 @@ def get_indexes_metadata(
 def get_col_type(col: Dict[Any, Any]) -> str:
     try:
         dtype = f"{col['type']}"
-    except Exception:  # pylint: disable=broad-except
+    except KeyError:
```

Comment:
Actually I think this one may be trickier. I was staring at this and wondering why a `KeyError` is raised and on the exception handling we are trying to access the same key again with `col["type"].__class__.__name__`. So the exception KeyError makes sense, because it may be possible, but what we are catching is a serialisation problem, `# sqla.types.JSON __str__ has a bug, so using __class__.`

Reply:
Thanks @dpgaspar for noticing this. Maybe I'm lacking the bigger picture here so excuse me if the question would be silly, but In that case why we are trying to do `f"{col['type']}"` instead just having `dtype = col["type"].__class__.__name__` without `try catch`?

Reply:
I think that we want the default to be a call to `sqla.type.__str__` and if that fails (on the comment it fails for `sqla.types.JSON`) then use the class name.

Reply:
Sure, then I'm reverting this change